torch_geometric/datasets/mesh_cnn_base_datasets.py

```diff
@@ -0,0 +1,480 @@
+                        (ind - 1) if (ind >= 0) else (len(pos) + ind)
+                        for ind in face_vertex_ids]
+                    faces.append(face_vertex_ids)
+            f.close()
+            faces = np.array(faces, dtype=int)
+        assert np.logical_and(faces >= 0, faces < len(pos)).all()
+        return Data(pos=pos, faces=faces)
+
+    def get_mean_std(self, sub_name=''):
+        """ Computes Mean and Standard Deviation from Training Data.
+        If mean/std file doesn't exist, will compute one
+        :returns
+        mean: N-dimensional mean
+        std: N-dimensional standard deviation
+        n_input_channels: N
+        (here N=5)
+        """
+        mean_std_cache = osp.join(self.processed_dir,
+                                  sub_name + 'mean_std_cache.p')
+        if not osp.isfile(mean_std_cache):
+            print('computing mean std from train data...')
+            # no augmentations during m/std computation
+            num_aug = self.num_aug
+            self.num_aug = 1
+            mean, std = np.array(0), np.array(0)
+            for j, mesh_data in enumerate(self):
+                if j % 500 == 0:
+                    print('{} of {}'.format(j, self.size))
+                features = mesh_data['edge_attr']
+                mean = mean + features.mean(axis=1)
+                std = std + features.std(axis=1)
+            mean = mean / (j + 1)
+            std = std / (j + 1)
+            transform_dict = {'mean': mean[:, np.newaxis],
+                              'std': std[:, np.newaxis],
+                              'ninput_channels': len(mean)}
+            with open(mean_std_cache, 'wb') as f:
+                pickle.dump(transform_dict, f)
+            print('saved: ', mean_std_cache)
+            self.num_aug = num_aug
+        # open mean / std from file
+        with open(mean_std_cache, 'rb') as f:
+            transform_dict = pickle.load(f)
+            print('loaded mean / std from cache')
+            self.mean = transform_dict['mean']
+            self.std = transform_dict['std']
+            self.n_input_channels = transform_dict['ninput_channels']
+
+
+class MeshCnnClassificationDataset(MeshCnnBaseDataset):
+    r"""This class creates a classification dataset for MeshCNN networks:
+    `"MeshCNN: A Network with an Edge" <https://arxiv.org/abs/1809.05910>`_
+    paper.
+    The dataset should be organized in folders according to class names. Each
+    class folder should have 'train' and 'test' folders, where on these folders
+    it should contain the .obj files. After the first pre-processing the class
+    will create 'cache' folders which contain pre-processed .npz files after
+    data augmentations.
+    Args:
+        root (str): Root folder for dataset.
+        dataset_url (str): dataset URL link (see supported links in
+                           MeshCnnBaseDataset)
+        n_input_edges (int): Number of input edges of mesh. It should be the
```

Comment:
Which one?
n_input_edges is there since MeshCNN must get the same number of edges in a batch. 
Therefore, n_input_edges is the maximal number of edges of a mesh object in the dataset. Any other mesh object with smaller number of input edges will be pad.